src/ff/prime_field.rs

```diff
@@ -1,5 +1,9 @@
 use super::{field::BinaryField, Field};
 use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Not};
+// something weird is happening and compiler does not see the use of this macro inside field_impl
+// but it happily complains if this import is removed.
+#[allow(unused_imports)]
+use paste::paste;
```

Comment:
Rust's approach to identifying macros is annoyingly inconsistent with how it identifies other things.  Otherwise, I would suggest you just change the below to "paste::paste! {".

But maybe we can avoid depending on that...

Reply:
Made it work without `paste` but it is a bit clunky.